test/performance/README.md

```diff
@@ -1,10 +1,81 @@
 
-Existing scenarios:
+## Adding performance tests
 
-- [Observed concurrency](./observed-concurrency)
+Performance tests are simple go tests, that use [Fortio](#load-generator) for generating 
+load and [Tesgrid](#displaying-metrics) to see the metrics on a continuous basis. For whitebox tests,
+the performance tests also bring up the prometheus service along with knative-serving. So, 
+you can [query prometheus](#prometheus-metrics) to get server side metrics with the load 
+generator metrics.
+
+## Load Generator
+
+Knative has a [load generator library](https://github.com/knative/test-infra/tree/master/shared/loadgenerator)
+that can be used to generate load. The load generator uses [Fortio](https://fortio.org/) for generating load based
+on the [generator options](https://github.com/knative/test-infra/blob/master/shared/loadgenerator/loadgenerator.go)
+set in the test. The load generator provides the [generator results](https://github.com/knative/test-infra/blob/master/shared/loadgenerator/loadgenerator.go)
+which includes all the data points for the load generation request and any calculated latencies.
+
+For eg.
+
+```go
+opts := loadgenerator.GeneratorOptions{Duration: 1*time.Minute, NumThreads: 1}
+resp, err := opts.RunLoadTest(false /* resolvableDomain */)
+if err != nil {
+	t.Fatalf("Generating traffic via fortio failed: %v", err)
+}
+```
+
+## Prometheus metrics
+
+Knative provides a [prometheus wrapper](https://github.com/knative/test-infra/tree/master/shared/prometheus) that
+provides methods to wait for prometheus to scrap for the metrics once the test is finished. It also provides a way
+to query the prometheus server for any server-side metrics and then display those in [testgrid](#displaying-metrics)
+
+For eg.
+```go
+promAPI, err := prometheus.PromAPI()
+if err != nil {
+	logger.Errorf("Cannot setup prometheus API")
+}
+query := fmt.Sprintf("%s{configuration_namespace=\"%s\", configuration=\"%s\", revision=\"%s\"}", metric, test.ServingNamespace, names.Config, names.Revision)
+val, err := prometheus.RunQuery(context.Background(), logger, promAPI, query)
+if err != nil {
+	logger.Infof("Error querying metric %s: %v", metric, err)
+}
+```
+
+## Displaying metrics
+
+Knative uses [testgrid](https://testgrid.knative.dev) to show all its metrics. Performance metrics are shown on a 
+separate tab for each repo like [serving](https://testgrid.knative.dev/knative-serving#performance). Once the test is
+done, each test can define which metrics they want to be captured and shown on testgrid. For each metric, create a 
+[testgrid testcase](https://github.com/knative/test-infra/blob/master/shared/testgrid/testgrid.go) by using the 
+[CreatePerfTestCase() method](https://github.com/knative/serving/blob/master/test/performance/performance.go). 
+
+For eg.
+
+```go
+testName := "TestPerformanceLatency"
+var tc []testgrid.TestCase
+for name, val := range metrics {
+	tc = append(tc, CreatePerfTestCase(val, name), testName))
```

Comment:
parenthesis misplaced. Also, maybe make the name of the arguments (in the code) more explicit? `tName` and `tcName` are kind of confusing. Otherwise explain the use of the parameters in the docstring.

Reply:
do you mean in https://github.com/knative/serving/blob/master/test/performance/performance.go#L99? I will update that in another PR. I want to keep this specific to README.

Reply:
The parenthesis are still misplaced.
`CreatePerfTestCase(val, name), testName)` should be `CreatePerfTestCase(val, name, testName)`